drv/cosmo-hf/src/main.rs

```diff
@@ -269,6 +269,7 @@ impl FlashDriver {
                     }
                 }
             }
+            offset += len as u32;
```

Comment:
should we be doing something to avoid a panic site from the overflow check here?